docs/coding-guidelines/add-types-to-SerDe.md

```diff
@@ -0,0 +1,100 @@
+
+Not every type has been implemented so if you need to use a type that isn't supported and you get the error "Type {0} not supported yet", you will need to implement the type.
+
+## Steps to follow
+
+### 1. Choose a new type identifier and add it to:
+
+https://github.com/dotnet/spark/blob/master/src/csharp/Microsoft.Spark/Interop/Ipc/PayloadHelper.cs#L20
+
+Example:
+
+```csharp
+
+    private static readonly byte[] s_doubleArrayArrayTypeId = new[] { ( byte)'A' };z
+
+```
+
+Once you have created your new identifier implement the csharp side writer in PayloadHelper.cs by 
+finding the method `ConvertArgsToBytes` and find out where in the switch that the code should be 
+implemented.
+
+This is how a byte array is written:
+
+```csharp
+
+    case double[][] argDoubleArrayArray:
+        SerDe.Write(destination, s_doubleArrayArrayTypeId);
+        SerDe.Write(destination, argDoubleArrayArray.Length);
+        foreach (double[] doubleArray in argDoubleArrayArray)
+        {
+            SerDe.Write(destination, doubleArray.Length);
+            foreach (double d in doubleArray)
+            {
+                SerDe.Write(destination, d);
+            }
+        }
+        break;
+
+```
+
+Typically the type identifier is written, then the length and then the data. Once you have written 
+the code that will perform the write you need to add a case to `GetTypeId` for your new identifier:
+
+```csharp
+
+    if (type == typeof(int[]) ||
+        type == typeof(long[]) ||
+        type == typeof(double[]) ||
+        type == typeof(double[][]) ||
+        typeof(IEnumerable<byte[]>).IsAssignableFrom(type) ||
+        typeof(IEnumerable<string>).IsAssignableFrom(type))
+    {
+        return s_arrayTypeId;
+    }
+```
+
+### 2. Derserialize this new type in JVM with read method
+
+In [SerDe.scala](https://github.com/dotnet/spark/blob/master/src/scala/microsoft-spark-2.3.x/src/main/scala/org/apache/spark/api/dotnet/SerDe.scala) (This need to be changed in all Spark version 2.3.x, 2.4.x and 3.0.x). 
+
+In either the method `readTypedObject` or `readList` add a new case statement for your new type 
+identifier:
+
+```scala
+
+  case 'A' => readDoubleArrArr(dis)
+
+```
+
+Once you have read the type, you can implement your method to read the data you wrote in csharp, examples:
+
+```scala
+
+   def readDoubleArrArr(in: DataInputStream): Array[Array[Double]] = {
+    val len = readInt(in)
+    (0 until len).map(_ => readDoubleArr(in)).toArray
+  }
+
+```
+
+That should be everything you need to be able to send a new type from the CLR to the JVM.
+
+### 3. Serialize this new type in JVM with write method
+
+In src/main/scala/org/apache/spark/api/dotnet/SerDe.scala the function `writeObject` contains a switch for each type, a couple of examples:
+
```

Comment:
How about adding `case "doublearray" => dos.writeByte('A')` also in this section?